### PR TITLE
 Makefile support for content generation and build/lint plugins by wildcard

### PR DESCRIPTION
**Pull request by scottyw**
~~There are two commits here.~~

This commit changes the Makefile to check for all directories in `cmd` that fit the pattern `goplugin-*` and builds/lints each on the assumption it contains a plugin. That should avoid Makefile changes as we (or contributors) add future plugins.

It also adds a new `generate` target that runs the bridge generator, rebuilds, regenerates types and sanity checks everything.

~~The second commit is all the changes to generated stuff that fell out of the 1st commit :)~~

~~@thallgren I included TS generation here. Good move do you think? Most of that 2nd commit is TS types.~~

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -32,22 +34,11 @@ shrink:
...
-	$(call build,goplugin-example,cmd/goplugin-example/main.go)
-	$(call build,goplugin-identity,cmd/goplugin-identity/main.go)
-	$(call build,goplugin-puppetdsl,cmd/goplugin-puppetdsl/main.go)
-	$(call build,goplugin-tf-aws,cmd/goplugin-tf-aws/main.go)
-	$(call build,goplugin-tf-azurerm,cmd/goplugin-tf-azurerm/main.go)
-	$(call build,goplugin-tf-github,cmd/goplugin-tf-github/main.go)
-	$(call build,goplugin-tf-google,cmd/goplugin-tf-google/main.go)
-	$(call build,goplugin-tf-kubernetes,cmd/goplugin-tf-kubernetes/main.go)
-
-puppet-dsl:
```
Comment (thallgren) [inline Makefile]: I would like to keep this actually. It saves me a lot of time when iterating changes in the framework using the puppet-dsl plugin.

**Comment by thallgren** [comment]
I think TS-generation is fine but I'm not in favor of the location, for two reasons:

1. Mixing type description in different languages in the same directory is bad. Why must I see puppet types clutter my types directory when I never use them?
2. TS is not a binary present under the "plugins" directory. What's the rationale of placing the TS-types there?

In some sense, it boils down to the name "plugins" not being ideal. We use that name because the only mechanism that we currently use is hc plugin infrastructure. The more we make things dependent on that, the more difficult it will get to change it moving forward.

**Comment by scottyw** [comment]
Let's drop TS. The generator put it in that dir by default but I agree with you that it makes sense to look at the bigger picture.

**Comment by scottyw** [comment]
Dropping TS for now and a rebase means there are now only Makefile changes.

File: Makefile (hunk 1 above)
Comment (scottyw) [reply Makefile]: Restored.

**Pull request merged**